src/main/java/com/mraof/minestuck/item/weapon/OnHitEffect.java

```diff
@@ -334,6 +336,32 @@ static OnHitEffect enemyKnockback(float knockback)
 		};
 	}
 	
+	/**
+	 * Flings the target in the direction the attacker was facing, and flings the attacker in the opposite direction of their facing
+	 */
+	static OnHitEffect mutualKnockback(float knockback)
+	{
+		return (itemStack, target, attacker) ->
+		{
+			Vec3 targetVec = attacker.getLookAngle().scale(1F + knockback);
+			Vec3 attackerVec = targetVec.reverse();
+			
+			target.push(targetVec.x, targetVec.y, targetVec.z);
+			
+			//dismount the attacker
+			if(attacker.getVehicle() != null)
+				attacker.dismountTo(attacker.getX(), attacker.getY(), attacker.getZ());
+			
+			attacker.push(attackerVec.x, attackerVec.y, attackerVec.z);
+			
+			if(attacker instanceof Player player)
```

Comment:
Might as well cast directly to `ServerPlayer` if you then need it for `sendToPlayer()`